internal/gitrepo/gitrepo.go

```diff
@@ -87,6 +90,8 @@ type RepositoryOptions struct {
 	CI string
 	// GitPassword is used for HTTP basic auth.
 	GitPassword string
+	// UseSSH flag to determine if HTTPS (default) or SSH will be used for git operations
+	UseSSH bool
```

Comment:
Instead of adding configuration to do this how about we just detect it? I think we can make use of `Remotes()` and maybe have a private method to check what type of remote it is. If the URL looks like an ssh we setup that auth, otherwise do what we have been doing. Let me know if I am missing something though where that won't be possible.

Reply:
Updated!